101-custom-rp-with-function/README.md

```diff
@@ -0,0 +1,96 @@
+
+```
+PUT/GET/DELETE /subscriptions/{subscriptionid}/resourceGroups/{resourcegroup}/providers/Microsoft.CustomProviders/resourceProviders/{customrpname}/users/name?<api-version>
+```
+
+You can see this in action in the last part of the template where we create this new user type : 
+
+```json
+        {
+            "type": "Microsoft.CustomProviders/resourceProviders/users",
+            "name": "[concat(parameters('funcname'), '/santa')]",
+            "apiVersion": "2018-09-01-preview",
+            "location": "eastus",
+            "properties": {
+                "FullName": "Santa Claus",
+                "Location": "NorthPole"
+            },
+            "dependsOn": [
+                "[concat('Microsoft.Web/sites/',parameters('funcname'))]"
+            ]
+        }
+```
+
+Navigating to the deployment details on the templatedeployment  will show a new resource type called resourceproviders/users created on the custom resource provider with the name santa
+![](images/createduser.png)
+
+To confirm that the user has been created , you can use a rest api client such as postman and run the following query and see the results:
+
+```
+GET  
+https://management.azure.com/subscriptions/{subscriptionid}/resourceGroups/{resourcegroup}/providers/Microsoft.CustomProviders/resourceProviders/{customrpname}/users/santa?api-version=2018-09-01-preview
+```
+
+You can also create\delete additional users.
+
+### Ping action
+
+In addition to users you can also define actions on your resourceprovider. An example action called ping is provided here defined as follows:
+
+```json
+"actions": [
+                                {
+                                    "name": "ping",
+                                    "routingType":"Proxy",
+                                    "endpoint": "[concat('https://', parameters('funcname'), '.azurewebsites.net/api/{requestPath}')]"
+                                }
+                            ]
+```
+
+The format for this action is similar to the resource type defined above , but since this is an action this call will only support POST methods as follows:
+
+```
+POST  
+https://management.azure.com/subscriptions/{subscriptionid}/resourceGroups/{resourcegroup}/providers/Microsoft.CustomProviders/resourceProviders/{customrpname}/ping?api-version=2018-09-01-preview
+```
+
+The code that enables this process is all implemented as part of the azure function that is deployed along with the template. To further understand how the function has been configured please look here:
+
++ [**Creating an azure function**](SampleFunctions/CSharpSimpleProvider/README.md)
```

Comment:
Update Link .. do you want to pull the functions readme here ?

Reply:
Not sure - it feels like we should just fold that into the main readme since it's part of the sample.  I'll merge and then let you guys decide how you want to handle that part...  (i.e. you can submit a change)

Reply:
cool I think thats ok for now , we can take a look and fix it later if required